src/bootstrap/test.rs

```diff
@@ -773,7 +773,9 @@ impl Step for RustdocGUI {
     const ONLY_HOSTS: bool = true;
 
     fn should_run(run: ShouldRun<'_>) -> ShouldRun<'_> {
-        run.path("src/test/rustdoc-gui")
+        let builder = run.builder;
+        let run = run.path("src/test/rustdoc-gui");
+        run.default_condition(builder.config.nodejs.is_some() && builder.config.npm.is_some())
```

Comment:
You should also make the "warning: npm is missing" line a hard error, or this doesn't help with making sure the test is run.

Reply:
True. Updating.